### ISSUE TITLE
SwitchFirst operator

### ISSUE THREAD

**Issue by jinroh** [ReactiveX/RxJS#645]
Is anyone working on the implementation of `SwitchFirst` right now ? If not I'd be happy to ! (I *really* need it in one of my project)

Some bikeshed I'd like to discuss before starting implementation:

- what about the name of the operator, since we dropped `switchLatest` in favor of `switchMap` (I don't think [the comment](https://github.com/ReactiveX/RxJS/issues/167#issuecomment-131260858) from @mattpodwysocki in issue #167 has been adressed). What about `switchFirst` and `switchMapFirst` to match `switch` and `switchMap` ?
- what about a concurrency parameter (see #347) ? if so, what signature exactly ?
- is this core or kitchensink ? I'm in favor of core...

**Comment by blesh** [comment]
I don't think anyone is working on it. I think you're right, though, the name needs work. `switchFirst` doesn't really describe what it's doing. I'm not sure what would though.

It subscribes to the first Observable it gets until it completes, dropping all others, then subscribes to the next one. It's almost a weird specialization of `sample` and `merge`. I've honestly never used it.

For now it could be named `switchFirst` and `switchFirstMap`... there just seems like there could be some more descriptive name or term for it.

Also, it can land in KitchenSink for now until we bikeshed what to do with it.

**Comment by blesh** [comment]
... also for now, let's skip the concurrency parameter until that can be discussed further. I'm not against it, I just haven't thought about it.

**Comment by kwonoj** [comment]
I think this can be closed now - @jinroh , may I possibly ask if there's additional PR might be required for this issue?

**Comment by blesh** [comment]
I don't think there is any more work to do here. If there is we can open another issue. Closing for now.
